scripts/lib/puppet_cache.py

```diff
@@ -50,28 +52,56 @@ def do_puppet_module_install(
     target_path: str,
     success_stamp: str,
 ) -> None:
-    # This is to suppress Puppet warnings with ruby 2.7.
-    distro_info = parse_os_release()
-    puppet_env = os.environ.copy()
-    if (distro_info["ID"], distro_info["VERSION_ID"]) in [("ubuntu", "20.04")]:
-        puppet_env["RUBYOPT"] = "-W0"
-
     os.makedirs(target_path, exist_ok=True)
     with open(PUPPET_DEPS_FILE_PATH) as yaml_file:
         deps = yaml.safe_load(yaml_file)
-        for module, version in deps.items():
-            run(
-                [
-                    "puppet",
-                    "module",
-                    "--modulepath",
-                    target_path,
-                    "install",
-                    module,
-                    "--version",
-                    version,
-                ],
-                env=puppet_env,
-            )
+    for module, metadata in deps.items():
+        install_puppet_module(target_path, module, metadata["version"], metadata["sha256sum"])
     with open(success_stamp, "w"):
         pass
+
+
+def install_puppet_module(
+    target_path: str, module: str, version: str, expected_sha256sum: str
+) -> None:
+    with urlopen(f"https://forgeapi.puppet.com/v3/releases/{module}-{version}") as forge_resp:
+        forge_data = json.loads(forge_resp.read())
```

Comment:
```suggestion
        forge_data = json.load(forge_resp)
```

(untested but I think this should work and avoid the string re-alloc. this is an optional nit)

Reply:
Clever idea.  Thanks, applied!